build.gradle

```diff
@@ -25,183 +21,165 @@ apply plugin: 'maven-publish'
 apply plugin: 'com.bmuschko.nexus'
 apply plugin: 'jacoco'
 apply from: custom('jacoco-version')
+apply from: custom('jacoco-config')
+apply from: custom('jacoco-excludes')
 
 sourceCompatibility = 1.8
+targetCompatibility = 1.8
 
 jar {
-	manifest {
-		attributes 'Implementation-Title': 'simple-react', 'Implementation-Version': version
-	}
+  manifest {
+    attributes 'Implementation-Title': 'simple-react', 'Implementation-Version': version
+  }
 }
 
-apply from: custom('jacoco-config')
-apply from: custom('jacoco-excludes')
-
-
 repositories {
-	mavenCentral()
-	jcenter()
-
+  mavenCentral()
+  jcenter()
 }
 
 if (JavaVersion.current().isJava8Compatible()) {
-	allprojects {
-		tasks.withType(Javadoc) {
-			options.addStringOption('Xdoclint:none', '-quiet')
-		}
-	}
+  allprojects {
+    tasks.withType(Javadoc) {
+      options.addStringOption('Xdoclint:none', '-quiet')
+    }
+  }
 }
 
-
-configurations {
-	provided
-}
 dependencies {
 
-
-
-	compile 'org.pcollections:pcollections:'+pCollectionsVersion
-	compile "org.jooq:jool:${joolVersion}"
-	compile 'org.agrona:Agrona:'+agronaVersion
-	compile 'org.reactivestreams:reactive-streams:'+reactiveStreamsVersion
-	provided group: 'org.projectlombok', name: 'lombok', version:"${lombokVersion}"
-	testCompile 'com.github.stefanbirkner:system-rules:1.10.0'
-	testCompile 'io.projectreactor:reactor-core:3.0.7.RELEASE'
-
-	testCompile  'com.google.guava:guava:19.0'
-	testCompile 'com.google.guava:guava-testlib:19.0'
-	testCompile 'org.reactivestreams:reactive-streams-tck:1.0.0'
-	testCompile group: 'junit', name: 'junit', version: '4.12'
-	testCompile group: 'org.hamcrest', name: 'hamcrest-all', version:hamcrestVersion
-	testCompile group: 'org.mockito', name: 'mockito-all', version:'1.9.5'
-	testCompile 'io.vertx:vertx-core:3.0.0'
-	testCompile group: 'org.testng', name: 'testng', version:'5.14.10'
-	testCompile 'io.javaslang:javaslang:2.0.2'
-	jmh 'com.github.akarnokd:ixjava:1.0.0-RC5'
-
-
+  compile 'org.pcollections:pcollections:2.1.2'
+  compile 'org.jooq:jool:0.9.12'
+  compile 'org.agrona:Agrona:0.9.1'
+  compile 'org.reactivestreams:reactive-streams:1.0.0'
+  compileOnly 'org.projectlombok:lombok:1.16.12'
```

Comment:
Introduced on gradle 2.12 it replaces the use of the **provided** hack used in previous versions.